src/main/java/jenkins/security/plugins/ldap/FromGroupSearchLDAPGroupMembershipStrategy.java

```diff
@@ -77,15 +77,15 @@ public void setAuthoritiesPopulator(LdapAuthoritiesPopulator authoritiesPopulato
     }
 
     @Override
-    public GrantedAuthority[] getGrantedAuthorities(LdapUserDetails ldapUser) {
-        return getAuthoritiesPopulator().getGrantedAuthorities(ldapUser);
+    public Collection<? extends GrantedAuthority> getGrantedAuthorities(DirContextOperations userData, String username) {
+        return getAuthoritiesPopulator().getGrantedAuthorities(userData, username);
     }
 
     @Override
-    public Set<String> getGroupMembers(String groupDn, LDAPConfiguration conf) throws DataAccessException {
+    public Set<String> getGroupMembers(String groupDn, LDAPConfiguration conf) {
         LDAPExtendedTemplate template = conf.getLdapTemplate();
         String[] memberAttributes = { "member", "uniqueMember", "memberUid" };
-        return (Set<String>) template.retrieveEntry(groupDn, new GroupMembersMapper(), memberAttributes);
+        return template.executeReadOnly(ctx -> new GroupMembersMapper().mapAttributes(groupDn, ctx.getAttributes(LdapUtils.getRelativeName(groupDn, ctx), memberAttributes)));
```

Comment:
Just guessing here. Read only? Read write? `retrieveEntry` is nowhere to be found in Spring Security so I tried to inline the Acegi Security behavior to the extent I understood it. Note that this means `GroupMembersMapper.mapAttributes` could just be inlined here too, but I am trying to keep the diff small where possible.

Reply:
All the calls should be read-only. The ldap plugin never writes in the remote ldap afair.

Reply:
> could just be inlined here too

⇒ #52